video/decode/vd_lavc.c

```diff
@@ -1116,6 +1156,9 @@ static bool decode_frame(struct dec_video *vd)
     mpi->pts = mp_pts_from_av(ctx->pic->pts, &ctx->codec_timebase);
     mpi->dts = mp_pts_from_av(ctx->pic->pkt_dts, &ctx->codec_timebase);
 
+    if (vd->vo->driver->get_buffer)
+        vd->vo->driver->flush_buffer(vd->vo, mpi);
```

Comment:
Makes sense too. Not sure if I'm happy about the additional callback, but I guess we can do it for now.